src/components/Modal/ServiceModal.js

```diff
@@ -93,6 +94,14 @@ const ServiceModal = ({ isOpen, onClose, onSubmit, CurrentNode, handleColorClick
                 [column]: value,
                 serverPort: value,
             }));
+        } else if (column === 'applicationFramework') {
+            if (value !== '') {
```

Comment:
can be changed to positive condition first and handle the negative case in else
if (value === ''){
   ...
} else {
  ....
}